agent/app/run.go

```diff
@@ -43,7 +49,14 @@ func Run(arguments []string) int {
 		// timeout of 30s. This is so that we can catch any http timeout and log the
 		// issue within agent logs.
 		// see https://docs.docker.com/engine/reference/builder/#healthcheck
-		return runHealthcheck("http://localhost:51678/v1/metadata", time.Second*25)
+		// if AGENT_HEALTHCHECK_LOCALHOST_OVERRIDE env var is set, it will override
+		// the healthcheck's localhost ip address inside the ecs-agent container
+		localhost := "localhost"
+		if localhostOverride := os.Getenv(HEALTHCHECK_LOCALHOST_OVERRIDE_ENV_VAR); localhostOverride != "" {
```

Comment:
we should probably parse this in the config package for consistency? https://github.com/aws/amazon-ecs-agent/blob/master/agent/config/config.go

Reply:
I didn't see a neat way to stuff config in this top level of the application.  I'll take another look.